### PR TITLE
Node compatability

### PR DESCRIPTION
**Pull request by leplatrem**
- [x] get rid of window export
- [x] use lib for Promise object
- [x] use lib for XMLHttpRequest object ([xhr](https://github.com/Raynos/xhr) ?)
- [x] update README

``` diff
 var utils = require('./ext/utils.js');
 var console = require("console");
+var Promise = require("bluebird");
+var XMLHttpRequest = require("xmlhttprequest").XMLHttpRequest;
 var hawk = require('hawk');
 var deriveHawkCredentials = require('./ext/hkdf.js').deriveHawkCredentials;
```

### CODE REVIEW

**Comment by leplatrem** [comment]
I wanted to run mocha in node too. But I can get _sinon_ and _fake http requests_ to work...

Especially `server.respondWith`. See https://github.com/spiral-project/daybed.js/compare/node_compatibility...test_node_sinon

We can postpone this precise problem for later and merge this though.

**Comment by almet** [comment]
see https://github.com/cjohansen/Sinon.JS/issues/319 for reference about using fake http requests with node/browserify.

**Comment by leplatrem** [comment]
Oh I've read this thread three times maybe, and played around with stuff I couldn't master. Feel free to catch up in the **test_node_sinon** branch!

File: README.md [hunk 1]
```diff
@@ -7,6 +7,21 @@ applications.
 [![Build Status](https://travis-ci.org/spiral-project/daybed.js.png?branch=master)](https://travis-ci.org/spiral-project/daybed.js)
 
 
+## Usage
+
+In the browser, just include the source file:
+
+```html
+  <script src="//spiral-project.github.io/daybed.js/build/daybed.js"></script>
```
Comment (almet) [inline README.md:15]: We probably should publish this at js.daybed.io/daybed.min.js, what do you think?

Comment (Natim) [inline README.md:15]: https://daybed.io/daybed.min.js could be enough if we want to serve it with AD but yes if we want to serve it with a CDN cdn.daybed.io is even better don't you think?

Comment (almet) [inline README.md:15]: (I'm thinking about using github CDN to serve our js minified files, like we're doing but 1. minifying them and 2. using our own domain so we don't tie our users to github).

File: gulpfile.js [hunk 2]
```diff
@@ -10,17 +10,20 @@ var deploy = require("gulp-gh-pages");
 
 var opt = {
   outputFolder: "build",
-
   app: {
-    src: ["./index.js", "./ext/utils.js", "./ext/hkdf.js"],
```
Comment (almet) [inline gulpfile.js:15]: We don't need utils and hkdf anymore?

File: README.md (hunk 1 above)
Comment (Natim) [inline README.md:15]: Yes let's go for js.daybed.io I like that :)

Comment (leplatrem) [inline README.md:15]: Note that daybed.js can't be minified currently.

File: gulpfile.js (hunk 2 above)
Comment (leplatrem) [inline gulpfile.js:15]: Yes, but there are scanned by browserify.

Comment (almet) [inline gulpfile.js:15]: ah ok cool.

File: README.md (hunk 1 above)
Comment (almet) [inline README.md:15]: What cannot be minified?

**Comment by almet** [comment]
That looks good to me!

File: README.md (hunk 1 above)
Comment (leplatrem) [inline README.md:15]: uglify() is not used in gulpfile currently.

**Pull request merged**